CodeEditModules/Modules/CodeFile/src/TextView/CodeEditorTextView.swift

```diff
@@ -89,4 +89,156 @@ public class CodeEditorTextView: NSTextView {
+        menu = removeTextMenu(menu)
+
+        return menu
+    }
+
+    func helpMenu(_ menu: NSMenu) -> NSMenu {
+        menu.insertItem(withTitle: "Jump To Definition",
+                        action: nil,
+                        keyEquivalent: "",
+                        at: 1)
+
+        menu.insertItem(withTitle: "Show Code Actions",
+                        action: nil,
+                        keyEquivalent: "",
+                        at: 2)
+
+        menu.insertItem(withTitle: "Show Quick Help",
+                        action: nil,
+                        keyEquivalent: "",
+                        at: 3)
+
+        menu.insertItem(.separator(), at: 4)
+
+        return menu
+    }
+
+    func codeMenu(_ menu: NSMenu) -> NSMenu {
+
+        menu.insertItem(withTitle: "Refactor",
+                        action: nil,
+                        keyEquivalent: "",
+                        at: 5)
+
+        menu.insertItem(withTitle: "Find",
+                        action: nil,
+                        keyEquivalent: "",
+                        at: 6)
+
+        menu.insertItem(withTitle: "Navigate",
+                        action: nil,
+                        keyEquivalent: "",
+                        at: 7)
+
+        menu.insertItem(.separator(), at: 8)
+
+        return menu
+    }
+
+    func gitMenu(_ menu: NSMenu) -> NSMenu {
+        menu.insertItem(withTitle: "Show Last Change For Line",
+                        action: nil,
+                        keyEquivalent: "",
+                        at: 9)
+
+        menu.insertItem(withTitle: "Create Code Snippet...",
+                        action: nil,
+                        keyEquivalent: "",
+                        at: 10)
+
+        menu.insertItem(withTitle: "Add Pull Request Discussion to Current Line",
+                        action: nil,
+                        keyEquivalent: "",
+                        at: 11)
+
+        menu.insertItem(.separator(), at: 12)
+
+        return menu
+    }
+
+    func removeSubMenus(_ menu: NSMenu) -> NSMenu {
+        // remove unwanted "Font" menu and its submenus
+
+        if let substitutionsItem = menu.item(withTitle: "Substitutions") {
+            menu.removeItem(substitutionsItem)
+        }
+
+        if let transformationsItem = menu.item(withTitle: "Transformations") {
+            menu.removeItem(transformationsItem)
+        }
+
+        if let speechItem = menu.item(withTitle: "Speech") {
+            menu.removeItem(speechItem)
+        }
+
+        if let shareItem = menu.item(withTitle: "Share") {
+            menu.removeItem(shareItem)
+        }
+
+        if let searchGoogleItem = menu.item(withTitle: "Search With Google") {
+            menu.removeItem(searchGoogleItem)
+        }
+
+        if let fontMenuItem = menu.item(withTitle: "Font") {
+            menu.removeItem(fontMenuItem)
+        }
+
+        return menu
+    }
+
+    func removeTextMenu(_ menu: NSMenu) -> NSMenu {
```

Comment:
How does this work/perform on non-english systems?
is it maybe better to not use the `super.menu(for: event)` to remove this.
This means that we must make a custom `cut`, `copy`, `paste`.